addons/sale_stock_margin/models/sale_order_line.py

```diff
@@ -15,7 +15,8 @@ def _compute_purchase_price(self):
             if not line.move_ids:
                 lines_without_moves |= line
             elif product.categ_id.property_cost_method != 'standard':
-                purch_price = product._compute_average_price(0, line.product_uom_qty, line.move_ids)
+                stock_moves = line.move_ids.with_company(line.company_id)
+                purch_price = product._compute_average_price(0, line.product_uom_qty, stock_moves)
```

Comment:
Since the variable isn't reused later on, we can just pass it directly

```suggestion
                purch_price = product._compute_average_price(0, line.product_uom_qty, line.move_ids.with_company(line.company_id))
```

Reply:
Yup we can pass it directly, but I thought it looks cleaner this way. Will do this change and add the test case as well.